tests/test_markers.py

```diff
@@ -303,6 +303,46 @@ def test_extra_with_no_extra_in_environment(self):
                 {"extra": "security"},
                 True,
             ),
+            (
+                "extra == 'security'",
+                {"extra": None},
+                False,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": None},
+                False,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "foobar"},
+                False,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7.0"},
+                True,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7"},
+                True,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7.0.0"},
+                True,
+            ),
+            (
+                "extra == '3.7.0'",
+                {"extra": "3.7.0+youaregreat"},
+                True,
```

Comment:
In PEP-440 `3.7.0==3.7.0+anything`, and `3.7.0==3.7.0.0` too.
That's how PEP-440 does the comparison, and how PEP-508 defines it, as it says it respects PEP-440 comparison for valid version specifier.
So,the marker `extra == '3.7.0'` with `extra=3.7.0+youaregreat` should evaluate as true.

We can open another issue about the PEP-508 spec and the changes we might want, but that's the "how PEP-508 is supposed to be right now" implementation.

Personnaly, I'm ok with that, and I'm eagerly waiting for the PEP-440 comparison to be fixed in the extras for pip, as I'd like to define some things like:
```
extras_require={
    [';extra~=3.7']: ['adependency']
    [';extra~=3.8']: ['anotherdependency']
}
```
Now, if I all my extras are valid version specifier, but I want `3.7.0` and `3.7.0.0` to be different, I could define those using `===`:
```
extras_require={
    [';extra===3.7.0']: ['adependency']
    [';extra===3.7.0.0']: ['anotherdependency']
}
```
One sad thing, if you have extras like `foobar`, `3.7.0` and `3.7.0.0`, you can't differentiate them all with PEP-508, as the `===` operator is not defined in python, so you can't use it here. That means you're back with `3.7.0==3.7.0.0`.
Like I said, that's a whole other discussion IMO.